reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt

```diff
@@ -634,6 +638,30 @@ class StaticHtmlReporter : Reporter {
     }
 }
 
+private fun DIV.licenseLink(license: String) {
+    val licenseResourcePath = getLicenseResourcePath(license)
+    val sha1Git = HashAlgorithm.SHA1_GIT.calculate(licenseResourcePath) ?: return +license
```

Comment:
The line potentially calculates the hash for the same license multiple times. Could you please add a final commit to this PR which uses a map via `getOrPut()` to reuse previously calculated `SHA1_GIT`s?